packages/frontend/src/redux/slices/handleAsyncThunkStatus.js

```diff
@@ -0,0 +1,41 @@
+import set from 'lodash.set';
+
+import initialErrorState from './initialErrorState';
+
+/**
+ * Automatically handle status part of reducer based on initialErrorState
+ *
+ * @param asyncThunk string representing async thunk type
+ * @param buildStatusPath array of strings representing the path to status state, needed because in a few cases we want to place the status object in more than one place
+ * @param builder objct provides addCase, addMatcher, addDefaultCase functions
+ */
+export default ({
+    asyncThunk,
```

Comment:
Will the async thunk object will always have properties for `pending`, `fulfilled`, and `rejected` like it does in the thunks being refactored in this PR? 

If so I think it makes more sense to pass the async thunk here rather than the type string. Then you don't need to repeat the action type when invoking `handleAsyncThunkStatus` and the typing becomes more explicit - i.e. an object with properties for `pending`, et al. rather than a string that can't be validated against known action types.

Reply:
It's not possible, unfortunately. I was trying to find a better solution, but it seems that there is none. 
`pending/fulfilled/rejected` are action creators, that are being created while creating the thunk. And matcher function in `addMatched` has `action` param only - so in this case, the only usable param to match is `action.type`, and we can't get that from the thunk function.

What you describe would be possible with `builder.addCase` but it's not allowed to use two of these in one slice for the same action creator, so we can only add one `addCase` for a particular action creator (pending/fulfilled/rejected). And in some cases, we are using `addCase` for different reasons than `status` object.

Reply:
Interesting, I saw your note on `addCase` in the PR description but hadn't made that connection.